packages/sdk/src/adapters/multipeer/rules.ts

```diff
@@ -257,35 +257,75 @@ export const Rules: { [id in Payloads.PayloadType]: Rule } = {
             beforeQueueMessageForClient: (
                 session: Session,
                 client: Client,
-                message: Message,
+                message: Message<Payloads.ActorCorrection>,
                 promise: ExportedPromise
             ) => {
-                message.payload.type = 'actor-update';
-                client.queueMessage(message, promise);
-                return undefined;
+                // Coalesce this actor correction with the previously queued update if it exists, maintaining a single
+                // update for this actor rather than queuing a series of them.  This is fine, as we do not need to lerp
+                // an actor correction on a late join user.  It can just be the updated actor values.
+                const payload = message.payload;
+                const queuedMessage = client.queuedMessages
+                    .filter(value =>
+                        value.message.payload.type === 'actor-update' &&
+                        (value.message.payload as Payloads.ActorUpdate).actor.id === payload.actorId).shift();
+                if (queuedMessage) {
+                    const existingPayload = queuedMessage.message.payload as Partial<Payloads.ActorUpdate>;
+                    existingPayload.actor = deepmerge(existingPayload.actor,  {
+                        payload: {
+                            actor: {
+                                transform: {
+                                    app: message.payload.appTransform
+                                }
+                            }
+                        }
+                    });
+                    return undefined;
```

Comment:
>return undefined; [](start = 20, length = 17)

Can you document why you're returning undefined here?

Reply:
I got this from the actor-update so I will update with a comment there as well.

---
In reply to: [294896385](https://github.com/microsoft/mixed-reality-extension-sdk/pull/350#discussion_r294896385) [](ancestors = 294896385)